packages/twenty-front/src/modules/page-layout/widgets/graph/utils/transformTwoDimensionalGroupByToLineChartData.ts

```diff
@@ -126,9 +134,57 @@ export const transformTwoDimensionalGroupByToLineChartData = ({
     seriesMap.get(seriesKey)!.set(xValue, aggregateValue);
   });
 
+  let sortedXValues = allXValues;
+
+  if (isDefined(configuration.primaryAxisOrderBy)) {
```

Comment:
Question: can we factorize any of this with bar chart ? Seems very similar but I would understand that we can't, although it should be extracted in a util too.